packages/react-positioning/src/types.ts

```diff
@@ -0,0 +1,190 @@
+
+  /**
+   * Position for the component. Position has higher priority than align. If position is vertical ('above' | 'below')
+   * and align is also vertical ('top' | 'bottom') or if both position and align are horizontal ('before' | 'after'
+   * and 'start' | 'end' respectively),
+   * then provided value for 'align' will be ignored and 'center' will be used instead.
+   */
+  position?: Position;
+
+  /**
+   * Enables the Popper box to position itself in 'fixed' mode (default value is position: 'absolute')
+   * @default false
+   */
+  positionFixed?: boolean;
+
+  /**
+   * Lets you displace a popper element from its reference element.
+   * This can be useful if you need to apply some margin between them or if you need to fine tune the
+   * position according to some custom logic.
+   */
+  offset?: Offset;
+
+  /**
+   * When the reference element or the viewport is outside viewport allows a popper element to be fully in viewport.
+   * "all" enables this behavior for all axis.
+   */
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  unstable_disableTether?: boolean | 'all';
+
+  /**
+   * Disables automatic repositioning of the component; it will always be placed according to the values of `align` and
+   * `position` props, regardless of the size of the component, the reference element or the viewport.
+   */
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  unstable_pinned?: boolean;
+
+  /**
+   * Applies max-height and max-width on popper to fit it within the available space in viewport.
+   * true enables this for both width and height.
+   * 'height' applies only `max-height` and 'width' for `max-width`
+   */
+  autoSize?: 'height' | 'width' | boolean;
+}
+
+export interface PopperProps extends PositioningProps {
+  /**
+   * Ref object containing the pointer node.
+   */
+  pointerTargetRef?: React.RefObject<Element>;
+
+  /**
+   * The content of the Popper box (the element that is going to be repositioned).
+   */
+  children: PopperChildrenFn | React.ReactElement;
+
+  /**
+   * If false, delays Popper's creation.
+   * @default true
+   */
+  enabled?: boolean;
+
+  /**
+   * Array of conditions to be met in order to trigger a subsequent render to reposition the elements.
+   */
+  positioningDependencies?: React.DependencyList;
+
+  /**
+   * Ref object containing the target node (the element that we're using as reference for Popper box).
+   */
+  targetRef: React.RefObject<Element> | PopperJs.VirtualElement;
+
+  /** Rtl attribute for the component. */
+  rtl?: boolean;
+}
+
+export interface PopperChildrenProps {
+  /**
+   * Popper's placement.
+   */
+  placement: PopperJs.Placement;
+
+  /**
+   * Function that updates the position of the Popper box, computing the new offsets and applying the new style.
+   */
+  scheduleUpdate(): void;
+}
+
+export type PopperShorthandProps = PositioningProps;
+
+export type PopperOptions = Omit<PopperProps, 'children' | 'targetRef'> & {
```

Comment:
I suggest to remove `PopperProps` and inline its interface there:

```tsx
export interface PopperOptions extends PositioningProps {
  // all props except `children` & `targetRef`
}
```

`PopperProps` have been used only for `Popper` component, but it's not there anymore :)